spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java

```diff
@@ -929,6 +937,7 @@ protected Collection<Message<?>> completeGroup(Message<?> message, Object correl
 			this.logger.debug(() -> "Completing group with correlationKey [" + correlationKey + "]");
 
 			result = this.outputProcessor.processMessageGroup(group);
+			Assert.notNull(result, "the group returned a null result");
```

Comment:
I think that is processor is responsible for null result, not group. But again: it really can return null. Might be something is off with our logic here?
Not saying to do that right now with such a huge pr: let’s just have a note to come back to this method!
For now, fix, please, this error message .